src/dealDashboard/dealDashboard.ts

```diff
@@ -60,4 +69,73 @@ export class DealDashboard {
   public get userCanAccessDashboard(): boolean {
     return !this.deal.isPrivate || this.deal.isRepresentativeUser || this.deal.isUserProposalLead;
   }
+
+  /**
+   * Executes the token swap. Called by the "Execute Token Swap" button on the UI
+   *  - pops up a modal to verify the user wants to execute the swap
+   *  - does nothing if they hit cancel
+   *  - if they hit "execute" it will execute and show the congrats modal
+   */
+  public async executeSwap(): Promise<void> {
+    const swapModal: IAlertModel = {
+      header: "Execute token swap",
+      message:
+        `<p>You are about to execute token swapping between the following two DAOs. Do you want to execute these swaps?</p>
+        <div class='modal-content'>${this.getDaoHtmlForSwap(this.deal.primaryDao)}${this.getDaoHtmlForSwap(this.deal.partnerDao)}</div>`,
+      buttonTextPrimary: "Execute Swap <i style='margin-left:5px;' class='fa'>&#xf021;</i>",
+      buttonTextSecondary: "Cancel",
+      buttons: ShowButtonsEnum.Both,
+      className: "executeSwap",
+    };
+    // show a modal confirming the user wants to execute the swap
+    const dialogResult = await this.alertService.showAlert(swapModal);
+    if (!dialogResult.wasCancelled) {
+      this.isExecutingSwap = true;
+      //the user said they wanted to execute the swap so call the swap contract
+      const transactionReceipt = await this.deal.execute();
+      if (transactionReceipt){
+        //if the swap succeeded, show the 'congrats' modal
+        const congratulatePopupModel: IAlertModel = {
+          header: "Congratulations!",
+          message: "<p class='excitement'>You have successfully executed the token swap!</p>",
+          confetti: true,
+          buttonTextPrimary: "Close",
+          className: "congratulatePopup",
+        };
+        await this.alertService.showAlert(congratulatePopupModel);
+      } else {
+        this.eventAggregator.publish("handleError", new EventConfig("There was an error while attempting to execute the token swap. Please try again later", EventMessageType.Info, "Execute Swap Error"));
+      }
+      this.isExecutingSwap = false;
+    }
+  }
+
+  /**
+   * Gets the HTML for the dao token swap modal popup
+   * @param dao
+   * @returns string
+   */
+  private getDaoHtmlForSwap(dao: IDAO): string {
```

Comment:
can we avoid this definition of HTML in the viewmodel?

Reply:
This HTML was moved to the modal component as requested